### PR TITLE
icu-14735 add search & pagination to aliases

### PR DESCRIPTION
**Pull request by lisbet-alvarez**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-14735

# Description

Add search & pagination to aliases.
NOTE: We will need to update db version so it doesn't break for users. (There is a draft pr to do this so we might want to revisit that PR and update it if neccessary).
There will still be some upcoming changes to ensure search & pagination does not break when there are 500k+ aliases.

## Screenshots (if appropriate)

https://github.com/user-attachments/assets/6b4d556b-8594-4c68-a49c-548500a09938

## How to Test

1. delete current indexedDB since alias index was added.
2. go to "Aliases"
3. experiment with searching & switching between pages.

## Checklist

- [X] I have added before and after screenshots for UI changes
- [ ] I have added JSON response output for API changes
- [ ] I have added steps to reproduce and test for bug fixes in the description
- [ ] I have commented on my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: addons/api/addon/services/indexed-db.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ export const modelIndexes = {
     '&id, attributes.created_time, attributes.type, attributes.name, attributes.description, attributes.scope.scope_id, attributes.plugin.name',
   'auth-method':
     '&id, attributes.created_time, attributes.type, attributes.name, attributes.description, attributes.is_primary, attributes.scope.scope_id',
+  alias:
+    '&id, attributes.created_time, attributes.type, attributes.value, attributes.name, attributes.description, attributes.destination_id, attributes.scope.scope_id',
 };
```
Comment (lisbet-alvarez) [inline addons/api/addon/services/indexed-db.js:34]: Since a change was made to the index we need to update the indexedDb version. I saw that there is already a draft PR for that. Perhaps reliable-workers pod can look into turning that into an open PR?

File: ui/admin/app/routes/scopes/scope/aliases/index.js [hunk 2]
```diff
@@ -4,10 +4,100 @@
...
+        ),
+      );
+      aliasesExist = await this.getAliasesExist(scope_id, totalItems);
+    }
+
+    return { aliases, aliasesExist, totalItems };
+  }
+
+  /**
+   * Sets aliasesExist to true if there exists any aliases.
```
Comment (DhariniJeeva) [inline ui/admin/app/routes/scopes/scope/aliases/index.js:78]:
```suggestion
   * Sets aliasesExist to true if there any aliases.
```

File: ui/admin/app/routes/scopes/scope/aliases/index.js [hunk 3]
```diff
@@ -4,10 +4,100 @@
...
+        aliases.map((alias) =>
+          hash({
+            alias,
+            target: alias.destination_id
+              ? this.store.findRecord('target', alias.destination_id)
+              : null,
+          }),
+        ),
+      );
+      aliasesExist = await this.getAliasesExist(scope_id, totalItems);
```
Comment (DhariniJeeva) [inline ui/admin/app/routes/scopes/scope/aliases/index.js:71]: is this to check if there are any aliases in that particular scope?

File: ui/admin/app/templates/scopes/scope/aliases/index.hbs [hunk 4]
```diff
@@ -18,155 +18,181 @@
...
-                (hash
-                  canUpdate=(can 'update model' B.data.alias)
-                  canDelete=(can 'delete model' B.data.alias)
-                )
-                as |perms|
-              }}
-                {{#if (or perms.canUpdate perms.canDelete)}}
-                  <Hds::Dropdown as |dd|>
+  <page.body class='search-filtering'>
+    {{#if @model.aliasesExist}}
```
Comment (DhariniJeeva) [inline ui/admin/app/templates/scopes/scope/aliases/index.hbs:29]: Hmm, can we name this differently? right now, `@model.aliases` and `@model.aliasesExist` read the same to me. Naming it with more context would tell us why we can't use `@model.aliases` to toggle the search bar visibility

**Comment by DhariniJeeva** [comment]
Thanks for introducing pagination to Aliases 👏  
I added a couple of comments, looks great otherwise!

File: ui/admin/app/routes/scopes/scope/aliases/index.js [hunk 5]
```diff
@@ -4,10 +4,100 @@
...
+      });
+      totalItems = aliases.meta?.totalItems;
+      // since we don't receive target info from aliases list API,
+      // we query the store to fetch target information based on the destination id
+      aliases = await Promise.all(
+        aliases.map((alias) =>
+          hash({
+            alias,
+            target: alias.destination_id
+              ? this.store.findRecord('target', alias.destination_id)
```
Comment (ZedLi) [inline ui/admin/app/routes/scopes/scope/aliases/index.js:66]: Do you think we should set `backgroundReload` to false to help prevent extra queries if it already exists in the store? I imagine we shouldn't need to try and refresh data that was already in the store, especially if people are going between pages

File: ui/admin/app/templates/scopes/scope/aliases/index.hbs [hunk 6]
```diff
@@ -18,155 +18,181 @@
...
+                  <Hds::Text::Body
+                    @tag='p'
+                    @size='100'
+                    @weight='medium'
+                    @color='faint'
+                  >
+                    {{t 'titles.not-set'}}
+                  </Hds::Text::Body>
+                {{/if}}
+              </B.Td>
```
Comment (ZedLi) [inline ui/admin/app/templates/scopes/scope/aliases/index.hbs:89]: I know this is a limitation of us not wanting to load all targets for the filter right now but without the filter, people are definitely going to think it's a bug that when they search the target name it doesn't filter down the table 😕

File: ui/admin/app/routes/scopes/scope/aliases/index.js (hunk 3 above)
Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/aliases/index.js:71]: yup correct

File: ui/admin/app/templates/scopes/scope/aliases/index.hbs (hunk 4 above)
Comment (cameronperera) [reply ui/admin/app/templates/scopes/scope/aliases/index.hbs:29]: Would something like `@model.doAliasesExist` read better? Try to make it more clear that this is a boolean compared to an array of alias models.

File: ui/admin/app/templates/scopes/scope/aliases/index.hbs (hunk 6 above)
Comment (cameronperera) [reply ui/admin/app/templates/scopes/scope/aliases/index.hbs:89]: For other resources that don't require a name, we display the resource ID. I know the ID isn't as helpful as a name but it could still be better than `Not Set`.

Comment (ZedLi) [reply ui/admin/app/templates/scopes/scope/aliases/index.hbs:89]: I think `Not Set` is different as there could be an alias that's not pointing to any target yet so it's in the absence of the target. `displayName` would still display the ID if the target name doesn't exist

File: ui/admin/app/templates/scopes/scope/aliases/index.hbs (hunk 4 above)
Comment (ZedLi) [reply ui/admin/app/templates/scopes/scope/aliases/index.hbs:29]: I will say we have been using this naming pattern across other pages so if we want to change it, we should apply it everywhere.

Comment (lisbet-alvarez) [reply ui/admin/app/templates/scopes/scope/aliases/index.hbs:29]: good points, i think `doAliasesExist` is clear. Ill also update in other places since its a simple change.

File: ui/admin/app/routes/scopes/scope/aliases/index.js (hunk 5 above)
Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/aliases/index.js:66]: good point, i didnt realize that was an option, originally thought that not includeing `reload: true` was sufficient to prevent extra queries.

File: ui/admin/app/templates/scopes/scope/aliases/index.hbs (hunk 6 above)
Comment (lisbet-alvarez) [reply ui/admin/app/templates/scopes/scope/aliases/index.hbs:89]: hmm good thinking, perhaps an `Hds::TooltipButton` in the `Points to` table header letting user know they can only search on id and not name? I'll reach out to design on this.

Comment (lisbet-alvarez) [reply ui/admin/app/templates/scopes/scope/aliases/index.hbs:91]: update: no changes are going to be made in regards to this, [slack thread ](https://hashicorp.slack.com/archives/C012GTH1C4E/p1727301278916459?thread_ts=1726778606.934579&cid=C012GTH1C4E) for context

File: ui/admin/tests/acceptance/aliases/list-test.js [hunk 7]
```diff
@@ -43,15 +50,32 @@ module('Acceptance | aliases | list', function (hooks) {
...
-
+    urls.alias = `${urls.aliases}/${instances.alias.id}`;
+    urls.aliasWithTarget = `${urls.aliases}/${instances.aliasWithTarget.id}`;
     intl = this.owner.lookup('service:intl');
 
     authenticateSession({});
   });
 
   test('users can navigate to aliases with proper authorization', async function (assert) {
+    this.server.schema.aliases.all().destroy();
```
Comment (cameronperera) [inline ui/admin/tests/acceptance/aliases/list-test.js:78]: non-blocking: Why are we destroying all aliases?

**Review by DhariniJeeva — APPROVED** [review]
looks great!

File: ui/admin/tests/acceptance/aliases/list-test.js (hunk 7 above)
Comment (lisbet-alvarez) [reply ui/admin/tests/acceptance/aliases/list-test.js:78]: Ah good question, those test cases also test the empty app state, before there were no aliases but for search & created some in `beforeEach` hook

File: ui/admin/app/routes/scopes/scope/aliases/index.js [hunk 8]
```diff
@@ -4,10 +4,102 @@
...
+import { inject as service } from '@ember/service';
+import { hash } from 'rsvp';
 
 export default class ScopesScopeAliasesIndexRoute extends Route {
+  // =services
+
+  @service store;
+  @service session;
+  @service can;
+  @service router;
```
[the remark this answers is not in the document]
Comment (lisbet-alvarez) [reply ui/admin/app/routes/scopes/scope/aliases/index.js:16]: ahh thank you, great catch!

**Review by cameronperera — APPROVED** [review]
LGTM

**Review by ZedLi — APPROVED** [review]
Looks good! 1M aliases is incredibly slow but at least it loads
